### ISSUE TITLE
Dataset Preprocessing Scripts

### ISSUE THREAD

**Issue by Chrixtar** [TencentARC/MotionCrafter#3]
Hi,

congratulations to this great work and thank you very much for this early code release!
I noticed that the dataset preprocessing scripts were not uploaded yet, if I am not mistaken. Please correct me if I am wrong.

It would be really helpful to have them as well to reproduce results.
Do you already have a plan when to release these?

Thank you in advance!

Best
Chris

**Comment by RuijieZhu94** [comment]
Hi Cris, thanks for your attention! Sorry for the late reply, I am still on holiday, later I will upload the data preprocessing code.

**Comment by maoxindong123** [comment]
Thanks for the amazing work! I'm also eagerly waiting for the preprocessing scripts. Also, just a quick heads-up: it seems only the determ model weights are available, and the diff version is still missing. Thanks again!

**Comment by RuijieZhu94** [comment]
Hi @Chrixtar @maoxindong123, the data preprocessing code is uploaded. For the Diffusion-version model weight, I need more time to check and get permission. Please use the determinstic version for now.

**Comment by Chrixtar** [comment]
Thank you very much for uploading the preprocessing code. One more question regarding the Synthia dataset: Which weather condition(s) (https://synthia-dataset.net/downloads/) do you use for the video sequences?
Would be nice to add this detail in your DATASET.md for reproducibility :)

**Comment by RuijieZhu94** [comment]
Hi @Chrixtar, I've now finished my internship and no longer have access to the training data. I believe I used scenarios with all weather conditions. Could you please check Table 8 to confirm?

<img width="516" height="453" alt="Image" src="https://github.com/user-attachments/assets/d0285b3d-5421-4bda-a21d-480f2c11d578" />

**Comment by RuijieZhu94** [comment]
Hi @maoxindong123, the Diff version is updated.

**Comment by Chrixtar** [comment]
Hi @RuijieZhu94 , thank you again for providing the scripts and additional details!
Do you happen to have also the preprocessing scripts for the additional evaluation datasets (sintel, monkaa, ddad(?)) as well as the configs for running the whole evaluation pipeline?
If I am not mistaken, there are some config files `configs/eval_datasets` missing.

Would be really great to be able to follow your evaluation setup :)

**Comment by RuijieZhu94** [comment]
Hi @Chrixtar, the evaluation code is uploaded.

**Comment by Chrixtar** [comment]
Thank you very much!
I have a few more questions that I will just ask here instead of opening a new ticket:
1. Which split have you used for evaluation on ScanNet++: `nvs_sem_val.txt`, `nvs_test_iphone.txt`, `nvs_test_iphone.txt`, `nvs_test.txt`, or `sem_test.txt`?
2. Have you used the train split of Sintel for evaluation (no problem, just double checking)? The default value for split in the `gen_sintel_video.py` is "training".
3. I am confused about the Spring statistics in the paper (49 videos) and also the fact that you train and evaluate on it. Please correct me if am wrong, but the dataset seems to only have 37 training and 10 test videos (so 47 videos). And I cannot manage to find any published depth / disparity and (optical) flow for the test split [here](https://darus.uni-stuttgart.de/dataset.xhtml?persistentId=doi:10.18419/darus-3376). Do you maybe create your own split?
4. In the `wmap_vaekl.py`, I have seen that there are unused loss weights `lambda_deform_mask` and `lambda_valid_mask` that are set to 0 in the configs, but also do not have any functionality in code. I am curious whether you have tried training the VAEs also for reconstruction of those masks or what were these weights for? Because currently, you only use these masks for loss masking, but I wondered what happens if there are test scenes with undefined depth, e.g., sky regions? What would MotionCrafter predict?

Thank you again for your help! :)

**Comment by RuijieZhu94** [comment]
1. We didn’t use the ScanNet++ for evaluation. This is an old entrance.
2. Yes, I seem to remember that only the training split had ground truth annotations. And our training data does not include Sintel.
3. When I saved the data, I saved each video in 150 frames, so the 37 videos were split into 49.
4. Good question. I originally wanted to try predicting masks, but time was tight and I didn't have time to fully validate it. I think predicting valid and deform masks is useful; you could try adding cross-entropy loss to see how it goes. In testing, we don't evaluate the geometry or motion of invalid regions.

**Comment by Chrixtar** [comment]
Thanks again for the quick answers!
One follow-up regarding my question 3 about the Spring dataset: Since to my knowledge, depth, and optical flow have not been published for the test split of Spring, how have you evaluated on it? Have you created your own split or have you used the original train split?

**Comment by RuijieZhu94** [comment]
@Chrixtar That's why we only use the original 37 videos in training split for training and testing:)

**Comment by Chrixtar** [comment]
Does that mean you created another split of the original training split or have you trained and evaluated on the same data for the Spring dataset? It would be really great if you could provide information on the splits that you have used for training and testing for each dataset, as I find it a bit difficult to understand your exact training and evaluation setup for datasets that have been used for both training and testing like Spring, Kubric, VKITTI2, Dynamic Replica, and Point Odyssey.

**Comment by RuijieZhu94** [comment]
@Chrixtar Yes, we usually extract first 25 frames of each video as the test split. Later I will ask my colleague to upload the split files.

**Comment by RuijieZhu94** [comment]
Hi @Chrixtar, I've uploaded the splits for your reference.

**Comment by Chrixtar** [comment]
Thanks again for uploading the splits.
In your metric calculation, did you treat points that are very far away form the camera in the background somehow differently?
Because looking at some evaluation examples from Kubric and Point Odyssey, for example, I would expect bpth the scale and shift point map alignment with ground truth as well as then the metric scores to be easily dominated by those points or am I missing something?
I have seen that there is the `use_weight` option in eval.py to `Use depth-based weight map during alignment`. Did you use that for the numbers in your paper? But even then, this would only alleviate the dominance of such points in the alignment, but not in the metric computation right?

**Comment by RuijieZhu94** [comment]
Hi @Chrixtar, Thanks for the question. We used `use_weight` in our geometric evaluation. This weight helps with better alignment and the scale of the GT, so I believe it has an impact on the evaluation metrics.

**Comment by Chrixtar** [comment]
Thanks for the quick answer! Only for the geometric evaluation or also for the motion evaluation?

**Comment by RuijieZhu94** [comment]
For Table2 only.

**Comment by Chrixtar** [comment]
Just to be sure: Not for the Geometry part in Table 1 or Table 3?

**Comment by RuijieZhu94** [comment]
Sorry. For Monkaa, DDAD, Sintel, exactly. So Table3 also enables `use_weight`.
